await-remote-artifacts/action.yml

```diff
@@ -0,0 +1,55 @@
+# Copyright 2023, Kry10 Limited
+# SPDX-License-Identifier: BSD-2-Clause
+
+name: Await workflow artifacts
+
+description: |
+  Wait for artifacts to become available in another workflow run,
+  and optionally download them.
+
+inputs:
+  repo:
+    required: true
+    description: |
+      The GitHub repository containing the artifacts, as `owner/repo-name`.
+
+  run-id:
+    required: true
+    description: |
+      The workflow run ID containing the artifacts.
+
+  artifact-names:
+    required: true
+    description: |
+      A space-separated list of artifact names to wait for.
+      Artifact names should only contain sensible characters,
+      and definitely no whitespace.
```

Comment:
Could we define this a bit closer to maybe `[0-9a-zA-Z_\-]` or something like that? Then we won't have to argue about sensible with our future selves :-)